class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+
+    // Permissions
+    protected $_capabilities_settings_page = 'gravityforms_razorpay';
+    protected $_capabilities_form_settings = 'gravityforms_razorpay';
+    protected $_capabilities_uninstall = 'gravityforms_razorpay_uninstall';
+
+    // Automatic upgrade enabled
+    protected $_enable_rg_autoupgrade = true;
+
+    private static $_instance = null;
+
+    const GF_RAZORPAY_KEY = 'gf_razorpay_key';
+    const GF_RAZORPAY_SECRET = 'gf_razorpay_secret';
+
+    public static function get_instance()
+    {
+        if (self::$_instance == null)
+        {
+            self::$_instance = new GFRazorpay();
+        }
+
+        return self::$_instance;
+    }
+
+    public function init_frontend()
+    {
+        parent::init_frontend();
+
+        add_action('gform_after_submission', array($this, 'pay_using_razorpay'), 10, 2);
+    }
+
+    public function plugin_settings_fields()
+    {
+        return array(
+            array(
+                'title'       => '',
+                'fields'      => array(
+                    array(
+                        'name'    => self::GF_RAZORPAY_KEY,
+                        'label'   => esc_html__( 'Razorpay Key', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'name'    => self::GF_RAZORPAY_SECRET,
+                        'label'   => esc_html__( 'Razorpay Secret', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__( 'Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        foreach ($this->get_customer_fields() as $field)
+        {
+            $field_id = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $field_id);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
+    }
+
+    public function get_customer_fields()
+    {
+        return array(
+            array('name' => 'first_name', 'label' => 'First Name', 'meta_name' => 'billingInformation_firstName'),
+            array('name' => 'last_name', 'label' => 'Last Name', 'meta_name' => 'billingInformation_lastName'),
+            array('name' => 'email', 'label' => 'Email', 'meta_name' => 'billingInformation_email'),
+            array('name' => 'address1', 'label' => 'Address', 'meta_name' => 'billingInformation_address'),
+            array('name' => 'address2', 'label' => 'Address 2', 'meta_name' => 'billingInformation_address2'),
```

Comment:
@captn3m0 @riagarg5 Not a big fan of this either. I think there should be a better way to do this. It's so easy to get lost trying to read through this array. 

I'd suggest moving this over to a different file, and indenting it so that it is more readable.